### PR TITLE
up until performance

### PR DESCRIPTION
**Pull request by reikookamoto**

### CODE REVIEW

**Comment by reikookamoto** [comment]
Attempted the first and last question for week 4 but struggled with the second one

File: okamoto-reiko/backup/compare-manifests.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import sys
+import pandas as pd
+
+def compare_manifests(file_a, file_b):
+    
+    # read in manifest A
+    csv_a = pd.read_csv(file_a)
```
Comment (gvwilson) [inline okamoto-reiko/backup/compare-manifests.py:7]: hadn't seen someone use pandas for this before - nice!

File: okamoto-reiko/backup/compare-manifests.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    manifest_b = {'name': csv_b['name'].values.tolist(),
+                  'hash': csv_b['hash'].values.tolist()}
+    
+    # define dictionary to store results
+    results = {'changed': [],
+               'renamed': [],
+               'deleted': [],
+               'added'  : []}
+    
+    for i in range(len(manifest_a['name'])):
```
Comment (gvwilson) [inline okamoto-reiko/backup/compare-manifests.py:22]: nicely laid out and commented- easy to follow

File: okamoto-reiko/backup/finddup.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    for file in filenames:
+        reader = open(file, "rb")
+        hash_out = hash_stream(reader)
+        
+        if not hash_out in results.keys():
+            results[hash_out] = [file]
+        else:
+            results[hash_out].append(file)
+            
+    dups = {k:v for (k,v) in results.items() if len(v) > 1} # only return duplicates
```
Comment (gvwilson) [inline okamoto-reiko/backup/finddup.py:28]: nice use of comprehensions

**Review by gvwilson — APPROVED** [review]
nicely done

**Pull request merged**